### DISCUSSION TITLE
npm v7 should stabilize a fair amount before/if being promoted to LTS

### DISCUSSION THREAD

**Discussion by DeeDeeG** [npm/feedback#331] [CLI]
Hi folks,

There is a roadmap item "cli: clarify LTS policy" (See: https://github.com/npm/roadmap/projects/1#card-54575168).

If there will be an LTS policy, I believe npm v7 needs more time to bake and stabilize before being promoted to LTS.

(In my dream scenario, v6 is designated as LTS, v7 is skipped as LTS to implicitly bless its "move fast break things" development approach as intentional, and to acknowledge its relatively fast and extensive churn compared to v6, and allow for more stability-focused development ahead of a v8 or v9, with louder and wider calls for testing and comment from the community than was done in the lead-up to npm v7. Again, in my dream scenario, there is a support window bridging over npm v7 to a future v8 that is more continuous with v6 than v7 is, or which has at least had more time for ideas to percolate and best solutions to be homed in on, compared with the relatively rapid process I observed from the outside looking in to npm v7.)

In the absence of any formal LTS policy, this conversation risks being premature. But on the contrary, I wanted to get this feedback out there in the spirit of being proactive and kicking the ball off earlier rather than later about process and public communication and notions of stability in npm.

I value the modernization and perf improvements in v7 greatly. I would just like to see a bit more "oomph" put into in the "stability" department as well, balancing the fast and exciting pace of development. v7 has been a bumpy ride for me, and I personally would like to see (and potentially in the open-source spirit contribute to) smoother sailing ahead for any future LTS npm release.

Thanks for considering. Best regards.

**Comment by ljharb** [comment]
https://github.com/npm/feedback/discussions/204 may be helpful.

**Reply by DeeDeeG** [reply]
Thanks! That definitely clears a lot up.

I think a couple of things are still confusing:
- The roadmap item "clarify LTS policy" seems a bit misleading, coming across that without knowing that (from #204): "in referencing our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy) we don’t actually have an LTS program at npm, it is an artifact of the pre npm 4 release." 
- From the [Support Policy](https://github.com/npm/cli/wiki/Support-Policy) wiki page, it seems the `lts` dist tag of `npm` releases on the registry was meant to be removed, but that hasn't happened. I suppose either the tag should be really deleted or the wiki page updated.
  - Per the wiki page: "The `lts` dist-tag, & any references in our documentation, are remnants that will be removed by: **February 28th, 2021**"
